odl/test/solvers/nonsmooth/adupdates_test.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2014-2017 The ODL contributors
+#
+# This file is part of ODL.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+
+"""Test for the alternating dual updates solver."""
+
+from __future__ import division
+import odl
+from odl.solvers.nonsmooth import adupdates_method
+from odl.solvers.nonsmooth.adupdates import adupdates_method_simple
+
+from odl.util.testutils import all_almost_equal
+
+
+# Places for the accepted error when comparing results
+HIGH_ACCURACY = 8
+LOW_ACCURACY = 4
+
+
+def test_adupdates():
+    """Test if the adupdates solver handles the following problem correctly:
+
+    ( 1  1/2 1/3 1/4) (x_1)   (a)
+    (1/2 1/3 1/4 1/5) (x_2)   (b)
+    (1/3 1/4 1/5 1/6) (x_3) = (c)
+    (1/4 1/5 1/6 1/7) (x_4)   (d).
+
+    The matrix is the ill-conditined Hilbert matrix, the inverse of which
+    can be given in closed form. If we set
+
+    (a)   (25/12)                    (x_1) = (1)
+    (b)   (77/60)                    (x_2) = (1)
+    (c) = (19/20)         then       (x_3) = (1)
+    (d)   (319/420),                 (x_4) = (1).
+
+    We solve the problem
+
+    min ||Ax - b||^2 + TV(x) s.t. x >= 0
+
+    for the matrix A, the r.h.s. b as above and the total variation TV, which
+    is given as the sum of the absoulute values of consecutive entries of the
+    solution. The solution of this problem is clearly x = (1, 1, 1, 1), since
```

Comment:
Well this is assuming that we're using periodic boundary conditions for TV (which is not default in ODL), but I guess we can let that pass for now :O

Reply:
Not really, this total variation takes three differences for a four-component vector. This is still not the standard behaviour, though, because in ODL the absolute value of the last element would be added to the functional (which was very irritating to most of my students).